File: 0088-merge-sorted-array/0088-merge-sorted-array.js
/**
 * @param {number[]} nums1
 * @param {number} m
 * @param {number[]} nums2
 * @param {number} n
 * @return {void} Do not return anything, modify nums1 in-place instead.
 */
var merge = function(nums1, m, nums2, n) {
    let i = 0, j = 0;
    while (n > 0) {
        if (nums1[m-i-1] >= nums2[n-1]) {
            nums1[nums1.length-j-1] = nums1[m-i-1];
            nums1[m-i-1] = 0;
            j++;
            i++;

        } else {
            nums1[nums1.length-j-1] = nums2[n-1];
            n--;
            j++;
        }
    }
    return nums1
};

File: 0088-merge-sorted-array/0088-merge-sorted-array.cpp
auto init = [](){
    ios::sync_with_stdio(0), cin.tie(0), cout.tie(0);
    return 'c';
}();
class Solution {
public:
    const void merge(vector<int>& nums1, const int m, const vector<int>& nums2, int n) {
        if (nums1.empty()) { nums1 = nums2; return; }
        if (nums2.empty()) return;
        int i = nums1.size() - 1;
        int n1Idx = nums1.size() - nums2.size() - 1;
        int n2Idx = nums2.size() - 1;
        while (i >= 0) {
            if (n1Idx == -1) {
                nums1[i] = nums2[n2Idx];
                --n2Idx;
            } else if (n2Idx == -1) {
                nums1[i] = nums1[n1Idx];
                --n1Idx;
            } else if (nums1[n1Idx] >= nums2[n2Idx]) {
                nums1[i] = nums1[n1Idx];
                --n1Idx;
            } else {
                nums1[i] = nums2[n2Idx];
                --n2Idx;
            }
            --i;
        }
        return;
    }
};